lib/archive/archiver.go

```diff
@@ -63,11 +68,20 @@ type Archiver interface {
 	// spec - describes filters on paths found in the data (include, exclude, rebase, strip)
 	// data - include file data in the tar archive if true, headers only otherwise
 	Export(op trace.Operation, store *url.URL, id, ancestor string, spec *FilterSpec, data bool) (io.ReadCloser, error)
+
+	// Import will process the input tar stream based on the supplied path spec and write the stream to the
+	// target device.
+	//
+	// store - the device store containing the target device
+	// id - the id for the device that is contained within the store
+	// spec - describes filters on paths found in the data (include, exclude, strip, target)
+	// tarstream - the tar stream that is to be written to the target on the device
+	Import(op trace.Operation, store *url.URL, id string, spec *FilterSpec, tarstream io.ReadCloser) error
 }
 
 // CreateFilterSpec creates a FilterSpec from a supplied map
 func CreateFilterSpec(op trace.Operation, spec map[string]FilterType) (*FilterSpec, error) {
-	var rebase, strip string
+	var rebase, strip, target string
```

Comment:
Just realized we can get rid of these guys and just use `fs.RebasePath`, `fs.StripPath`, `fs.TargetPath` once we create the filterspec below. We can check in the switch against those instead of these vars.